api/src/main/java/org/eclipse/microprofile/jwt/builder/JwtClaimsBuilder.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.builder;
+
+import java.util.Set;
+
+/**
+ * JWT Claims Builder.
+ * 
+ * <p>
+ * JwtClaimsBuilder implementations must set the 'iat' (issued at time), 'exp' (expiration time)
```

Comment:
So when we set the iat in the call to Jwt.claims(Map<>) call, it needs to use that value and ignore the one the developer sets through JwtClaimsBuilder. issuedAt().

I would revert that logic or even disallow setting 'registered' claims through the Jwt.claims(Map<>) call.

Reply:
@rdebusscher Sorry, don't think so. There is no ambiguity. If the builder is set with some claim then unless it has been reset with a `claim` or typed claim setter it remains as is, This doc mentions that that if no `iat` is available then it must be provided OOB for the users to avoid doing the boilerplate code when the defaults suit.

Reply:
Ok, that was not clear for me from the Javadoc. (and my comment wasn't clear either I think, sorry)

So when a value is set by the initial creation with the map `Jwt.claims(Map<>)`, it will and can be overridden by a call to `JwtClaimsBuilder. issuedAt().`

If that is the expected behavior, I prefer this also.